api/spec/requests/spree/api/v2/storefront/products_spec.rb

```diff
@@ -401,6 +403,40 @@
           end
         end
       end
+
+      context 'sorting by sku' do
+        let(:skus_array) { json_response['data'].map { |p| p['attributes']['sku']} }
+
+        context 'ascending order' do
+          before { get '/api/v2/storefront/products?sort=sku' }
+
+          it_behaves_like 'returns 200 HTTP status'
+
+          it 'returns products sorted by master variant sku' do
+            expect(skus_array).to eq(skus_array.sort)
+            expect(json_response['data'].count).to      eq store.products.available.count
+            expect(json_response['data'].pluck(:id)).to eq store.products.available.joins(:master).
+              where("#{Spree::Variant.table_name}" => { is_master: true }).
+              select("#{Spree::Product.table_name}.*, #{Spree::Variant.table_name}.sku").
+              order("#{Spree::Variant.table_name}.sku").map(&:id).map(&:to_s)
+          end
+        end
+
+        context 'descending order' do
+          before { get '/api/v2/storefront/products?sort=-sku' }
+
+          it_behaves_like 'returns 200 HTTP status'
+
+          it 'returns products sorted by master variant sku with descending order' do
+            expect(skus_array).to eq(skus_array.sort.reverse)
+            expect(json_response['data'].count).to      eq store.products.available.count
+            expect(json_response['data'].pluck(:id)).to eq store.products.available.joins(:master).
```

Comment:
@szymoniwacz let's not use Arel queries in tests, we need to type down the ids by hand to perfectly sure :)